spec/PeriodicBackgroundSync-index.bs

```diff
@@ -110,3 +112,97 @@ The user agent should cap the duration and frequency of these events to limit re
+
+The <dfn attribute>periodicSync</dfn> attribute's getter must return the [=context object=]'s [=ServiceWorkerRegistration/periodic sync manager=].
+</div>
+
+## {{PeriodicSyncManager}} interface ## {#periodicsyncmanager-interface}
+<script type="idl">
+[Exposed=(Window,Worker)]
+interface PeriodicSyncManager {
+    Promise<void> register(DOMString tag, optional BackgroundSyncOptions options);
+    Promise<sequence<DOMString>> getTags();
+    Promise<void> unregister(DOMString tag);
+};
+
+dictionary BackgroundSyncOptions {
+    [EnforceRange] unsigned long long minInterval = 0;
+};
+</script>
+
+A {{PeriodicSyncManager}} has a <dfn attribute for=PeriodicSyncManager>service worker registration</dfn> (a [=service worker registration=]).
+
+The <code><dfn method for=PeriodicSyncManager title="register(tag, options)">register(|tag|, |options|)</dfn></code> method, when invoked, MUST return <a>a new promise</a> |promise| and run the following steps <a>in parallel</a>:
+
+1. Let |serviceWorkerRegistration| be the {{PeriodicSyncManager}}'s associated {{PeriodicSyncManager/service worker registration}}.
+1. If |serviceWorkerRegistration|’s [=active worker=] is null, [=reject=] |promise| with an {{InvalidStateError}} and abort these steps.
+1. if the {{PermissionState}} for a {{PermissionDescriptor}} with {{PermissionDescriptor/name}} <code>"periodic-background-sync"</code> is not {{granted}}, [=reject=] |promise| with a {{NotAllowedError}} and abort these steps.
+1. Let |currentRegistration| be the [=periodicsync registration=] in |serviceWorkerRegistration|'s [=list of periodicsync registrations=] whose [=tag=] equals |tag| if it exists, else null.
+1. If |currentRegistration| is null:
+    1. Let |newRegistration| be a new [=periodicsync registration=].
+    1. Set |newRegistration|'s associated [=tag=] to |tag|.
+    1. Set |newRegistration|'s associated [=options=] to |options|.
+    1. Set |newRegistration|'s associated [=state=] to [=pending=].
+    1. Set |newRegistration|'s associated {{periodicsync registration/service worker registration}} to |serviceWorkerRegistration|.
+    1. Add |newRegistration| to |serviceWorkerRegistration|'s [=list of periodicsync registrations=].
+    1. [=Calculate a time to fire=] for |newRegistration|.
+    1. [=Resolve=] |promise|.
+1. Else:
+    1. If the |currentRegistration|'s [=options=] are different from |options|:
+        1. Set |currentRegistration|'s associated [=options=] to |options|.
+        1. [=Calculate a time to fire=] for |newRegistration|.
+        1. Set |currentRegistration|'s associated [=state=] to [=pending=].
+    1. Else, if |serviceWorkerRegistration|'s [=registration state=] is [=firing=], set |serviceWorkerRegistration|'s [=registration state=] to [=reregisteredWhileFiring=].
```

Comment:
Isn't registration tied to the periodicsync registration rather than the serviceWorkerRegistration?

Reply:
good point. fixed.